bundles/org.pgcase.xobot.dbproc.jdbc/src/org/pgcase/xobot/dbproc/jdbc/functions/JdbcFunctionExtractor.java

```diff
@@ -0,0 +1,52 @@
+package org.pgcase.xobot.dbproc.jdbc.functions;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+
+import org.pgcase.xobot.dbproc.jdbc.DbprocJdbc;
+import org.pgcase.xobot.dbproc.runtime.XIssueReporter;
+import org.pgcase.xobot.dbproc.runtime.functions.XFunctionDescriptor;
+import org.pgcase.xobot.dbproc.runtime.functions.XFunctionExtractor;
+
+public class JdbcFunctionExtractor implements XFunctionExtractor {
+	private static final String SCAN_DBPROC_FUNC_SQL_FILE = "/sql/dbproc_jdbc_functions.sql"; //$NON-NLS-1$
+	private static String sqlSentence = null;	
+
+	@Override
+	public Iterable<XFunctionDescriptor> extractFunctions(Object input, Map<String, Object> context, XIssueReporter reporter) {
+		if (input instanceof Connection) {
+			try (Connection jdbcConnection = (Connection) input) {		
```

Comment:
This will close the connection after the block, we agreed to not close it here

Reply:
jdbcConnection  - clone from input. Connection will not close, will lost context only. Original input will not touch except use for select.

Reply:
I know about reference. I know that it's Java!
I debugged it. Connection created outside extractFunctions was exists and correct after extractFunctions exit. Really.
No need chjanges.